code/parse/parselo.cpp

```diff
@@ -3807,6 +3812,108 @@ int split_str(const char *src, int max_pixel_w, SCP_vector<int> &n_chars, SCP_ve
 	return line_num;
 }
 
+// A narrower but much faster alternative to split_str(), takes a string and a max pixel length, returns a vector with
+// one string per line. Does not currently support a max line count or ignoring of characters.
+SCP_vector<SCP_string>
```

Comment:
This should return a collection of offsets and lengths, not a list of strings.  This avoids copying the strings (multiple times, in this particular implementation) or erasing string characters (again, multiple times), therefore improving the algorithm's efficiency.  A vector of `std::pair<size_t, size_t>` would work well.

Reply:
I think this would be a bad change. In strict terms yes what you describe is the more optimal method, however, the split method was only ever a hot path due to the old method being stupendously wasteful. The overhead these changes would remove accounts for a miniscule amount of frame time, and I have tested unrealistic extreme use cases to confirm that.

Yes, client code of the old method is built around that return type. And frankly, I think that makes the client code harder to read. I think it is an example of excessively, unjustifiably low-level code that makes many parts of the engine harder to work with than needed. I would only want my code to handle strings that way if it was within a type that hid that complexity away.

To put it bluntly, rewriting and retesting this just so that I can go from extreme cases seeing a 5-6ms speedup to, optimistically, a 5.1-6.1ms speedup, is not a use of my time I am interested in.